### PR TITLE
WAPM publishing

### PR DESCRIPTION
**Pull request by jcaesar**
Implements https://github.com/ikanago/omekasy/issues/29.

I've tested this on some other repository, but I obviously can't test it on yours. So there might be some quirks.

For the login token, you'll have to:
* Log in to wapm.io,
* click your user account icon (top right),
* Settings,
* Access Tokens,
* Create new Token.

And then register that as a new repository secret on this github repo.

### CODE REVIEW

**Comment by jcaesar** [comment]
Actually, can I ask you to wait with merging this for a bit? It might be necessary to downgrade rust to 1.56 for the release build because of this [bug](https://github.com/wasmerio/webassembly.sh/issues/112).

**Comment by ikanago** [comment]
OK, please notify me after the bug is fixed.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -38,31 +39,33 @@ jobs:
           echo "PROJECT_VERSION=${GITHUB_REF#refs/tags/v}" >> "$GITHUB_ENV"
           echo "Version: ${{ env.PROJECT_VERSION }}"
 
+      - name: Check git tag against toml values
+        if: ${{ matrix.job.os != 'windows-2019' }} # windows too weird
```
Comment (ikanago) [inline .github/workflows/release.yml:43]: These files are identical among jobs, so just checking on one job, `ubuntu-20.04` for example, is enough.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -38,31 +39,33 @@ jobs:
...
+        run: |
+          grep -q "version = \"$PROJECT_VERSION\"" Cargo.toml
+          grep -q "version = \"$PROJECT_VERSION\"" wapm.toml
+
       - name: Build release
-        run: cargo build --locked --verbose --release --target=${{ matrix.job.target }}
+        run: cargo build --locked --verbose --release --target=${{ matrix.job.target }} --no-default-features --features '${{ matrix.job.features }}'
+        env:
+          RUSTFLAGS: -C strip=symbols
```
Comment (ikanago) [inline .github/workflows/release.yml:51]: I didn't know this option, but I've checked this reduces binary size. Thanks!

**Review by ikanago — CHANGES_REQUESTED** [review]
After the bug is fixed, I will test these workflows.

File: .github/workflows/release.yml (hunk 2 above)
Comment (jcaesar) [reply .github/workflows/release.yml:51]: (Only reason I used it is that I didn't want to have to install wabt.)

File: .github/workflows/release.yml (hunk 1 above)
Comment (jcaesar) [reply .github/workflows/release.yml:43]: That is what I thought at first, too, but tell me: Won't the publish step for the other `os` still run, creating a release with some missing files?
https://github.com/ikanago/omekasy/blob/a36cd66cc002d59effa34fadf311d7957a842d1c/.github/workflows/release.yml#L89-L95

**Comment by jcaesar** [comment]
I give up. Instead of that bug above getting fixed, webassembly.sh is getting more and more broken. You could of course publish to wapm.io for other reasons, but I'm not quite sure what those would be.

(Unrelated: I might package this for nixpkgs, though. [Edit:] [Easy](https://github.com/NixOS/nixpkgs/pull/325744).)

**Pull request closed without merging**